examples/sysparam_editor/sysparam_editor.c

```diff
@@ -41,12 +42,12 @@ void usage(void) {
 
 size_t tty_readline(char *buffer, size_t buf_size, bool echo) {
     size_t i = 0;
-    int c;
+    char c;
 
     while (true) {
-        c = getchar();
+        read(0, (void *)&c, 1);
         if (c == '\r' || c == '\n') {
-            if (echo) putchar('\n');
+            if (echo) printf("\n");
```

Comment:
Can not see why it would make any difference using `read` versus `getchar`, or `printf` versus `putchar`.

The comments on this suggest adding the following to the `Makefile`
```
EXTRA_COMPONENTS=extras/stdin_uart_interrupt
```

The problem might even be elsewhere as the wdt should not be triggered just because `getchar` spins. I think it needs more analysis to find the real problem. I have not been able to reproduce this issue, and do use this editor.